### PR TITLE
Add example configuration for IDEA as a merge tool

### PR DESCRIPTION
**Pull request by HybridEidolon**
Adds an additional example configuration for IntelliJ IDEA alongside kdiff3 and vimdiff.

# Checklist

If applicable:
- [x] I have updated the documentation (README.md, docs/, demos/)

### CODE REVIEW

**Comment by ilyagr** [comment]
I put a few more details in https://github.com/martinvonz/jj/wiki/Jetbrains-IntelliJ-IDEs.

I think we could link that wiki from `config.md`. Alternatively, if we had a predefined config like you sugest, we'd want to mention it there, and explain that the user will probably have to change the `program` field.

File: docs/config.md [hunk 1]
```diff
@@ -547,6 +547,10 @@ merge-tools.vimdiff.merge-args = ["-f", "-d", "$output", "-M",
     "-c", "set write"]
 merge-tools.vimdiff.program = "vim"
 merge-tools.vimdiff.merge-tool-edits-conflict-markers = true    # See below for an explanation
+
+# For IntelliJ IDEA or other JetBrains products on macOS
+merge-tools.idea.program = "/Applications/IntelliJ IDEA.app/Contents/MacOS/idea"
```
Comment (Cretezy) [inline docs/config.md:552]: We could add a note that for Linux, the launcher script must be installed and that this value must be pointed to it.

https://www.jetbrains.com/help/idea/working-with-the-ide-features-from-command-line.html

**Review by Cretezy — APPROVED** [review]

**Comment by HybridEidolon** [comment]
Given that there _is_ a wiki for things like this, maybe this doesn't need to go into the documentation? I wasn't aware the github wiki had anything on it.

**Comment by ilyagr** [comment]
I don't think this should go into the documentation as-is. If anything, we could change the documentation to make the wiki easier to discover. (E.g. a mention that there is a config for IDEA there).

**Pull request closed without merging**